base/gmp.jl

```diff
@@ -69,6 +69,96 @@ function __init__()
     end
 end
 
+
+module MPZ
+# wrapping of libgmp functions
+# "output parameters" are labeled x, y, z, and are returned when appropriate
+# constant input parameter are labeled a, b, c
+
+using Base.GMP: BigInt, Limb
+
+gmpz(op::Symbol) = (Symbol(:__gmpz_, op), :libgmp)
+
+sizeinbase(a::BigInt, b) = Int(ccall((:__gmpz_sizeinbase, :libgmp), Csize_t, (Ptr{BigInt}, Cint), &a, b))
+
+for op in (:add, :sub, :mul, :fdiv_q, :tdiv_q, :fdiv_r, :tdiv_r, :gcd, :lcm, :and, :ior, :xor)
+    @eval begin
+        $op(x::BigInt, a::BigInt, b::BigInt) = (ccall($(gmpz(op)), Void, (Ptr{BigInt}, Ptr{BigInt}, Ptr{BigInt}), &x, &a, &b); x)
```

Comment:
There is also the "problem" that since we are extending `Base.gcd`, there will also be method `gcd(::BigInt, ::BigInt, ::BigInt)` exposed to the user, which is does funny things from a user perspective. In particular it is breaking the following behaviour:
```julia
julia> gcd(BigInt(3), BigInt(2), BigInt(4))
1
```
With the proposed change the function mutates the first argument and returns `2`.

I don't think this is what we want and `gcd` is not the only function with this problem.